syntax = "proto2";

package py4mac;

import "generic.proto";
import "item.proto";

option py_generic_services = true;

service RestApi {

  rpc getItem (common.GetItem) returns (common.GetItem.Response) {
    option (rpc) = {
      endpoints: [{
        method: "GET",
        path: "/items/{id}",
        since { major: 2, minor: 0 },
      }],
      visibility: PUBLIC,
      rpc_doc_title: "Get Item by Id",
    };
  }

  rpc getItems (common.GetItems) returns (common.GetItems.Response) {
    option (rpc) = {
      endpoints: [{
        method: "GET",
        path: "/items",
        since { major: 2, minor: 0 },
      }],
      visibility: PUBLIC,
      rpc_doc_title: "Get Items",
    };
  }

  rpc postItem (common.PostItem) returns (common.PostItem.Response) {
    option (rpc) = {
      endpoints: [{
        method: "POST",
        path: "/items",
        since { major: 2, minor: 0 },
      }],
      visibility: PUBLIC,
      rpc_doc_title: "Create Item",
    };
  }
}